spring-batch-infrastructure/src/test/java/org/springframework/batch/item/data/builder/MongoItemReaderBuilderTests.java

```diff
@@ -0,0 +1,166 @@
+	@Mock
+	private MongoOperations template;
+
+	private Map<String, Sort.Direction> sortOptions;
+
+	private ArgumentCaptor<Query> queryContainer;
+
+	@Before
+	public void setUp() throws Exception {
+		MockitoAnnotations.initMocks(this);
+		this.sortOptions = new HashMap<>();
+		this.sortOptions.put("name", Sort.Direction.DESC);
+		this.queryContainer = ArgumentCaptor.forClass(Query.class);
+	}
+
+	@Test
+	public void testBasic() throws Exception {
+		MongoItemReader<String> reader = getBasicBuilder().build();
+
+		when(template.find(this.queryContainer.capture(), eq(String.class))).thenReturn(new ArrayList<>());
+
+		assertNull("reader should not return result", reader.read());
+
+		Query query = this.queryContainer.getValue();
+		assertEquals(50, query.getLimit());
+		assertEquals(0, query.getSkip());
+		assertEquals("{ }", query.getQueryObject().toJson());
+		assertEquals("{ \"name\" : -1 }", query.getSortObject().toJson());
+	}
+
+	@Test
+	public void testFields() throws Exception {
+		MongoItemReader<String> reader = getBasicBuilder().fields("{name : 1, age : 1, _id: 0}").build();
+
+		when(this.template.find(this.queryContainer.capture(), eq(String.class))).thenReturn(new ArrayList<>());
+
+		assertNull("reader should not return result", reader.read());
+
+		Query query = this.queryContainer.getValue();
+		assertEquals(1, query.getFieldsObject().get("name"));
+		assertEquals(1, query.getFieldsObject().get("age"));
+		assertEquals(0, query.getFieldsObject().get("_id"));
+	}
+
+	@Test
+	public void testHint() throws Exception {
+		MongoItemReader<String> reader = getBasicBuilder().hint("{ $natural : 1}").build();
+
+		when(this.template.find(this.queryContainer.capture(), eq(String.class))).thenReturn(new ArrayList<>());
+
+		assertNull("reader should not return result", reader.read());
+
+		Query query = this.queryContainer.getValue();
+		assertEquals("{ $natural : 1}", query.getHint());
+	}
+
+	@Test
+	public void testCollection() throws Exception {
+		MongoItemReader<String> reader = getBasicBuilder().parameterValues(new ArrayList<Object>() {
+			{
+				add("foo");
+			}
+		}).query("{ name : ?0 }").collection("collection").build();
+
+		ArgumentCaptor<String> collectionContainer = ArgumentCaptor.forClass(String.class);
+
+		when(this.template.find(this.queryContainer.capture(), eq(String.class), collectionContainer.capture()))
+				.thenReturn(new ArrayList<>());
+
+		assertNull("reader should not return result", reader.read());
+
+		Query query = this.queryContainer.getValue();
+		assertEquals("{ \"name\" : \"foo\" }", query.getQueryObject().toJson());
+		assertEquals("{ \"name\" : -1 }", query.getSortObject().toJson());
+		assertEquals("collection", collectionContainer.getValue());
+	}
+
+	@Test
+	public void testNullTemplate() {
+		validateExceptionMessage(new MongoItemReaderBuilder<String>().targetType(String.class).query("{ }")
+				.sorts(this.sortOptions).name("mongoReaderTest").pageSize(50), "template is required.");
```

Comment:
Formatting for all of the builder expressions in this class.  Will fix on merge.